index.cjs

```diff
@@ -0,0 +1,41 @@
+/*eslint-env node*/
+'use strict';
+
+var path = require('path');
+
+// If in 'production' mode, use the combined/minified/optimized version of Cesium
+if (process.env.NODE_ENV === 'production') {
+    module.exports = require(path.join(__dirname, 'Build/Cesium/Cesium'));
+    return;
+}
+
+try {
+    module.exports = require('esm')(module)('./Source/Cesium.js');
+} catch (e) {
+    if (e.code !== 'ERR_REQUIRE_ESM') {
+        throw e;
+    }
+
+    // Node 12+ throws an exception when trying to load an ES module via `esm`,
+    // because `esm` loads Cesium ES modules via require instead of import, and
+    // that is an illegal thing to do because Cesium's package.json declares
+    // `"type": "module"`. Node doesn't know that `esm` is magically
+    // transforming the ES modules to CommonJS.
+    //
+    // So, we override the loader for .js files to skip the exception.
+    // Idea from here:
+    // https://github.com/standard-things/esm/issues/855#issuecomment-566825957
+    var fs = require('fs');
+
+    var originalFunc = require.extensions['.js'];
+    require.extensions['.js'] = function(module, filename) {
```

Comment:
This is a global change to the node environment, right? Sounds like we don't have a lot of options, just wanted to confirm.

Just to throw out an alternative, we could always use the built version of Cesium in NodeJS and by requiring  `CesiumUnminified/Cesium.js` we would still get extra debugging and such.

Reply:
Actually, if `esm` is going the way of the dodo, maybe it's better to just use combined in all cases now and we can revisit going full import later on (Node 14?)

Reply:
Great idea! I got rid of the hack and made `index.cjs` always load the built version, either `Build/Cesium/Cesium.js` or `Build/CesiumUnminified/Cesium.js` depending on the `NODE_ENV`.

I also hooked up sourcemaps so it still looks like separate source file while debugging. I couldn't get it to work in the minified build for some reason, though. Uglify/Rollup are supposed to be able to do sourcemapping in this scenario, but the generated sourcemaps were rubbish. Step into a Cesium function in a release build and you'd end up somewhere totally random instead of in the function you're actually in.

If you use Node 12 and proper ES module support, you'll use the ES modules directly, with no transpilation by `esm` or other shenanigans. However, the debug pragmas won't be removed in this scenario. That's unfortunate, but a problem for another day, I think.

Reply:
Awesome, thanks again for your work on this.  We talked offline, but just to summarize for anyone else, `rollup-plugin-strip-pragma` can be used for pragma removal, debug sourcemaps are still enabled, release will have to wait for a different PR to figure out the problem.